src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderServiceTest.java

```diff
@@ -37,4 +40,38 @@ void createSnapshotRequest() {
             datasetId, SnapshotBuilderTestData.createSnapshotAccessRequest(), email),
         response);
   }
+
+  @Test
+  void enumerateSnapshotRequestsByDatasetId() {
+    UUID datasetId = UUID.randomUUID();
+    SnapshotAccessRequestResponse responseItem = new SnapshotAccessRequestResponse();
+    List<SnapshotAccessRequestResponse> response = List.of(responseItem, responseItem);
+    when(snapshotRequestDao.enumerateByDatasetId(datasetId)).thenReturn(response);
+    Assertions.assertEquals(
+        snapshotBuilderService.enumerateByDatasetId(datasetId),
+        snapshotBuilderService.convertToEnumerateModel(response));
+  }
+
+  @Test
+  void convertToEnumerateModel() {
+    SnapshotAccessRequestResponse inputItem =
+        SnapshotBuilderTestData.createSnapshotAccessRequestResponse();
+    List<SnapshotAccessRequestResponse> response = List.of(inputItem);
+
+    EnumerateSnapshotAccessRequestItem expectedItem = new EnumerateSnapshotAccessRequestItem();
+    expectedItem.id(inputItem.getId());
+    expectedItem.status(inputItem.getStatus());
+    expectedItem.createdDate(inputItem.getCreatedDate());
+    expectedItem.name(inputItem.getSnapshotName());
+    expectedItem.researchPurpose(inputItem.getSnapshotResearchPurpose());
+    EnumerateSnapshotAccessRequest expected = new EnumerateSnapshotAccessRequest();
+    expected.addItemsItem(expectedItem);
```

Comment:
You could consolidate these statements which IMO makes them a little easier to read:

```
EnumerateSnapshotAccessRequestItem expectedItem =
  new EnumerateSnapshotAccessRequestItem()
    .id(inputItem.getId())
    .status(inputItem.getStatus())
    .createdDate(inputItem.getCreatedDate())
    .name(inputItem.getSnapshotName())
    .researchPurpose(inputItem.getSnapshotResearchPurpose());
EnumerateSnapshotAccessRequest expected =
  new EnumerateSnapshotAccessRequest().addItemsItem(expectedItem);
```